docs/data/sdk-middleware.md

```diff
@@ -0,0 +1,130 @@
+---
+
+Middleware lets you extend Amplitude by running a sequence of custom code on every event. This pattern is flexible and you can use it to support event enrichment, transformation, filtering, routing to third-party destinations, and more.
+
+!!!note
+    Middleware is only supported in maintance SDK except the maintance Browser SDKand legacy Ampli. **[Plugins](../sdk-plugins/)** replaced middleware in the latest SDK or latest Ampli version.
+
+## Middleware Structure
+ 
+### Function Signature 
+
+Each middleware is a simple function with this signature:
+
+```js
+/**
+ * A function to run on the Event stream (each logEvent call)
+ *
+ * @param payload The `payload` contains the `event` to send and an optional `extra` that lets you pass custom data to your own middleware implementations.
+ * @param next Function to run the next middleware in the chain, not calling next will end the middleware chain
+ */
+
+function (payload: MiddlewarePayload: next: MiddlewareNext): void;
+```
+
+???types "Types in  Middleware"
+    |<div class="med-column">Name</div>| Type | Description |
+    | - | - | - |
+    | `MiddlewarePayload.event` | Event | The event data being sent. The event may vary by platform. |
+    | `MiddlewarePayload.extra` | { [x: string]: any } | Unstructured object to let users pass extra data to middleware. |
+
+To invoke the next middleware in the queue, use the `next` function. You must call `next(payload)` to continue the middleware chain. If a middleware doesn't call `next`, then the event processing stops executing after the current middleware completes.
+
+Add middleware to Ampli via `amplitude.addEventMiddleware()`. You can add as many middleware as you like. Each middleware runs in the order in which it's added.
+
+### Payload Customization
+
+Middleware access to event fields may vary by platform. To ensure comprehensive access, we recommend updating to the latest Ampli version and utilizing the [Plugins](../sdk-plugins) feature.
+
+For browser ampli, the following are the accessable keys under `payload`.
+
+|<div class="med-column">Name</div>|Type|
+| - | - |
+| `event.event_type` | string |
+| `event.event_properties` | { [key: string]: any } |
+| `event.user_id` | string |
+| `event.device_id` | string |
+| `event.user_properties` | { [key: string]: any } |
+| `extra` | { [x: string]: any } |
+
+For other platforms, middleware can access and modify the entire Event JSON object, allowing for comprehensive adjustments as needed. Learn more at [here](../../../analytics/apis/http-v2-api/#keys-for-the-event-argument).
+
+## Middleware examples
+
+Use an Middleware to modify event properties, transformation, filtering, routing to third-party destinations, and more:
+!!!example
+
+    ???code-example "Drop-event middleware example(click to expand)"
```

Comment:
What about renaming this "Filtering middleware example"?

Missing space before `(click`